modules/social_features/social_content_report/src/EventSubscriber/FlagSubscriber.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace Drupal\social_content_report\EventSubscriber;
+
+use Drupal\Core\Cache\CacheTagsInvalidatorInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Entity\EntityStorageException;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Logger\LoggerChannelTrait;
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Drupal\flag\Event\FlagEvents;
+use Drupal\flag\Event\FlaggingEvent;
+
+/**
+ * Class FlagSubscriber.
+ */
+class FlagSubscriber implements EventSubscriberInterface {
+
+  use StringTranslationTrait, LoggerChannelTrait;
+
+  /**
+   * Whether to unpublish the entity immediately on reporting or not.
+   *
+   * @var bool
```

Comment:
I think will be better to leave it as boolean value but change data type in all other places where this configuration using because for this value using checkbox field.

Reply:
@OH17L In the future we want to change this to a number instead of a boolean. This is why the data storage is also currently an integer instead of a boolean. We should mark this field as an integer as well because that's what we get handed. If we treat it like a boolean but in the future suddenly get an integer then things may break.